Bottom: 4663b73541337c68073acb89dcf35d5407545ee9
Top:    bae4ddc1e280471886b38e134fd6d42c57ada52a
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-05-17 17:29:18 +0800

Refresh of virtio-net-correctly-redirect

---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index f34794a..59702f9 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -800,7 +800,7 @@ static struct sk_buff *receive_mergeable(struct net_device *dev,
 			}
 			*xdp_xmit = true;
 			if (unlikely(xdp_page != page))
-				goto err_xdp;
+				put_page(page);
 			rcu_read_unlock();
 			goto xdp_xmit;
 		default:
